File: 1 - metodosBusca/pacoteBusca_JomiHubner/src/grafo/Grafo.java

package grafo;

import java.util.*;

public abstract class Grafo {
	Map<Integer,Vertice> vertices = new HashMap<Integer,Vertice>();
	
	private void addVertice(Vertice v) {
		vertices.put(v.getId(),v);
	}
	
	public void criaVertice(int id) {
		addVertice(new Vertice(id));
	}
    
    public Vertice getVertice(int id) {
        return vertices.get(id);
    }
	
	public abstract void criaAresta(int i, int j);
    public abstract void criaAresta(int i, int j, int custo);
	
	public String toString() {
		StringBuffer out = new StringBuffer();
		for (Vertice v: vertices.values()) {
			out.append(v.getId() + ": " + v.getAdjacentes() + "\n");
		}
		return out.toString();
	}
}

File: 1 - metodosBusca/python/app/core/grafo/grafo.py
from abc import ABC, abstractmethod
from .vertice import Vertice

class Grafo(ABC):
    def __init__(self):
        self.vertices = {}

    def add_vertice(self, v):
        self.vertices[v.get_id()] = v

    def cria_vertice(self, id):
        self.add_vertice(Vertice(id))

    def get_vertice(self, id):
        return self.vertices.get(id)

    @abstractmethod
    def cria_aresta(self, i, j):
        pass

    @abstractmethod
    def cria_aresta(self, i, j, custo):
        pass

    def __str__(self):
        out = []
        for v in self.vertices.values():
            out.append(f"{v.get_id()}: {v.get_adjacentes()}")
        return "\n".join(out)